File: java_sdk/src/main/java/business_api_client/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * The set of filters for each inclusion rule.
 */
@Schema(description = "The set of filters for each inclusion rule.")
@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-03-21T10:49:20.864324-07:00[America/Los_Angeles]")
public class DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet {
  @JsonProperty("filters")
  private List<DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters> filters = new ArrayList<DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters>();

  @JsonProperty("operator")
  private String operator = null;

  public DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet filters(List<DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters> filters) {
    this.filters = filters;
    return this;
  }

  public DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet addFiltersItem(DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters filtersItem) {
    this.filters.add(filtersItem);
    return this;
  }

   /**
   * An array of filters on previous audience actions and optionally also on URL keywords or parameters. Max size: 1.
   * @return filters
  **/
  @Schema(required = true, description = "An array of filters on previous audience actions and optionally also on URL keywords or parameters. Max size: 1.")
  public List<DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters> getFilters() {
    return filters;
  }

  public void setFilters(List<DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters> filters) {
    this.filters = filters;
  }

  public DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet operator(String operator) {
    this.operator = operator;
    return this;
  }

   /**
   * Operator between filters in the filters object array. Enum value: OR. Within each inclusion rule, you can only specify one object in filters.
   * @return operator
  **/
  @Schema(required = true, description = "Operator between filters in the filters object array. Enum value: OR. Within each inclusion rule, you can only specify one object in filters.")
  public String getOperator() {
    return operator;
  }

  public void setOperator(String operator) {
    this.operator = operator;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet dmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet = (DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet) o;
    return Objects.equals(this.filters, dmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.filters) &&
        Objects.equals(this.operator, dmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.operator);
  }

  @Override
  public int hashCode() {
    return Objects.hash(filters, operator);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet {\n");
    
    sb.append("    filters: ").append(toIndentedString(filters)).append("\n");
    sb.append("    operator: ").append(toIndentedString(operator)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters} from './DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters.js';

/**
 * The DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet model module.
 * @module model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet
 * @version 1.2.1
 */
export class DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet {
  /**
   * Constructs a new <code>DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet</code>.
   * The set of filters for each inclusion rule.
   * @alias module:model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet
   * @class
   * @param filters {Array.<module:model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters>} An array of filters on previous audience actions and optionally also on URL keywords or parameters. Max size: 1.
   * @param operator {String} Operator between filters in the filters object array. Enum value: OR. Within each inclusion rule, you can only specify one object in filters.
   */
  constructor(filters, operator) {
    this.filters = filters;
    this.operator = operator;
  }

  /**
   * Constructs a <code>DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet} obj Optional instance to populate.
   * @return {module:model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet} The populated <code>DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet();
      if (data.hasOwnProperty('filters'))
        obj.filters = ApiClient.convertToType(data['filters'], [DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters]);
      if (data.hasOwnProperty('operator'))
        obj.operator = ApiClient.convertToType(data['operator'], 'String');
    }
    return obj;
  }
}

/**
 * An array of filters on previous audience actions and optionally also on URL keywords or parameters. Max size: 1.
 * @member {Array.<module:model/DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSetFilters>} filters
 */
DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.prototype.filters = undefined;

/**
 * Operator between filters in the filters object array. Enum value: OR. Within each inclusion rule, you can only specify one object in filters.
 * @member {String} operator
 */
DmpcustomAudiencerulecreateRuleSpecInclusionRuleSetFilterSet.prototype.operator = undefined;

